File: GreenSignal/Data/Models/IncidentReportAttribute.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Data.Models
{
    public class IncidentReportAttribute
    {
        [Key]
        public string Id { get; set; }

        [Required]
        public Guid IncidentReportId { get; set; }

        public IncidentReport IncidentReport { get; set; }             

        [Required]
        public string Name { get; set; }

        public string? StringValue { get; set; }

        public double? NumberValue { get; set; }

        public bool? BoolValue { get; set; }
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/IncidentReportAttribute.kt
package com.example.greensignal.domain.model.response

import com.example.greensignal.data.remote.dto.response.IncidentReportAttributeDto

data class IncidentReportAttribute(
    val id: String = "",
    val name: String = "",
    val stringValue: String? = null,
    val numberValue: Double? = null,
    val boolValue: Boolean? = null
)

fun IncidentReportAttributeDto.toIncidentReportAttribute(): IncidentReportAttribute {
    return IncidentReportAttribute(
        id = id,
        name = name,
        stringValue = stringValue,
        numberValue = numberValue,
        boolValue = boolValue
    )
}
